app/src/main/java/com/example/campusguide/map/displayIndoor/ChangeFloor.kt

```diff
@@ -0,0 +1,90 @@
+package com.example.campusguide.map.displayIndoor
+
+import android.view.View
+import com.example.campusguide.map.GoogleMapAdapter
+import kotlin.collections.HashMap
+
+class ChangeFloor constructor(private val map : GoogleMapAdapter): View.OnClickListener {
+
+    private val buildings = buildBuildings(map)
+    private var currentBuilding: String = ""
+    private var currentFloor = -1
+
+    companion object ButtonInfo {
+        var upButtonId = -1
+        var downButtonId = -1
```

Comment:
are these supposed to have the same value? i feel like the later check of v?.id == up or down ButtonId would never reach the downButtonId portion of the code?

Reply:
These are temporary values just so that I can declare the property. 

They are later assigned to the id value of the button view. This is because onChangeFloor is an onClickListener that listens to **both** the up and down buttons so that I can share information between them such as the floor plans and which floor is the current floor.

Reply:
ahh i see thank u!!